front/app/containers/Admin/reporting/components/ReportBuilderPage/EmptyState.test.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import EmptyState from './EmptyState';
+import { render, screen } from 'utils/testUtils/rtl';
+import * as FeatureFlag from 'hooks/useFeatureFlag';
+
+const mockFeatureFlag = FeatureFlag as { default: () => boolean };
+
+describe('EmptyState', () => {
+  it('renders with an enabled button to create a report when the feature flag is on', () => {
+    mockFeatureFlag.default = jest.fn(() => true);
+    const onOpenModal = jest.fn();
+    render(<EmptyState onOpenModal={onOpenModal} />);
+
+    const createReportButton = screen.getByRole('button', {
+      name: 'Create a report',
+    });
+    expect(createReportButton).not.toBeDisabled();
+  });
+
+  it('renders with a disabled button when the feature flag is turned off', () => {
```

Comment:
```suggestion
  it('renders with a disabled button when the feature is not allowed by the pricing plan', () => {
```

Reply:
Technically, we are testing based on the feature flag. Adding that meaning here means that if that changes this test name needs to change. I'd prefer not to do that and let the usage show instead. Same as above

Reply:
Okay 👍